src/features/query-bar/components/filter/Condition.js

```diff
@@ -47,46 +47,28 @@ const Condition = ({
 
     const updateColumnField = (option: OptionType) => {
         const { value: columnId } = option;
-
         onColumnChange(condition, columnId);
     };
 
-    const updateSelectedField = (option: OptionType, fieldType?: string) => {
-        const { type, value } = option;
-
-        let keyType = '';
-
-        switch (fieldType) {
-            case OPERATOR:
-                keyType = OPERATOR_KEY;
-                break;
-            case VALUE:
-                keyType = VALUE_KEY;
-                break;
-            default:
-                throw new Error('invalid input');
-        }
-
-        onFieldChange(condition, value, keyType, type);
+    const updateOperatorField = (option: OptionType) => {
+        const { value } = option;
+        onFieldChange(condition, value, OPERATOR);
     };
 
-    const getFormattedOptions = (options: Array<Object>): any[] => {
-        return options.map(option => {
-            const { displayName, key, type } = option;
-            return {
-                displayText: displayName,
-                type, // TODO: valueType
-                value: key || displayName, // we want key from COLUMN_OPERATORs when setting the value of the operator that was selected
-            };
-        });
+    const updateSelectedField = (option: OptionType) => {
```

Comment:
Maybe there's a better name for this. Its main use is to detect the new option that was selected for a `SingleSelectField`.